CoreFindView/src/au/gov/asd/tac/constellation/views/find2/plugins/advanced/AdvancedSearchPlugin.java

```diff
@@ -0,0 +1,717 @@
+    }
+
+    /**
+     * This function checks to see if a graph elements color attribute matches
+     * the criteria specified by the ColourCriteriaValues.
+     *
+     * @param values the color criteriaValues
+     * @param attributeInt the int of the attribute
+     * @param currElement the currentElement
+     * @param graph the current graph
+     * @return
+     */
+    private boolean searchAsColor(final FindCriteriaValues values, final int attributeInt, final int currElement, final GraphWriteMethods graph) {
+        final ColourCriteriaValues colorValues = (ColourCriteriaValues) values;
+        final ConstellationColor color = graph.getObjectValue(attributeInt, currElement);
+        boolean matches = false;
+
+        // if the color of the attribute matches the users color matches = true
+        if (colorValues.getFilter().equals("Is") && colorValues.getColorValue().equals(color)) {
+            matches = true;
+            // if the color of the attribute doesnt match the users color matches = true
+        } else if (colorValues.getFilter().equals("Is Not") && !colorValues.getColorValue().equals(color)) {
+            matches = true;
+        }
+        return matches;
+
+    }
+
+    /**
+     * This function checks to see if a graph elements zoned date time attribute
+     * matches the criteria specified by the DateTimeCriteriaValues.
+     *
+     * @param values the date time criteriaValues
+     * @param attributeInt the int of the attribute
+     * @param currElement the currentElement
+     * @param graph the current graph
+     * @return
+     */
+    private boolean searchAsDateTime(final FindCriteriaValues values, final int attributeInt, final int currElement, final GraphWriteMethods graph) {
+        DateTimeCriteriaValues dateTimeValues = (DateTimeCriteriaValues) values;
+        final String dateTimeString = graph.getStringValue(attributeInt, currElement);
+        boolean matches = false;
+
+        // convert the string date time into a zonedDateTime for comparisons
+        String[] splitDateTime = dateTimeValues.getDateTimeStringPrimaryValue().split(" ");
+        String parseFormatedString = splitDateTime[0] + "T" + splitDateTime[1] + splitDateTime[2];
+        ZonedDateTime valueDateTime = ZonedDateTime.parse(parseFormatedString);
+
+        // convert the string date time into a zonedDateTime for comparisons
+        String[] splitAttributeDateTime = dateTimeString.split(" ");
+        String parseAttributeFormatedString = splitAttributeDateTime[0] + "T" + splitAttributeDateTime[1] + splitAttributeDateTime[2];
+        ZonedDateTime attributeDateTime = ZonedDateTime.parse(parseAttributeFormatedString);
+
+        /**
+         * This switch handles the matching check based on the filter choice by
+         * the user. It checks the users date time values against the attributes
+         * date time values
+         */
+        switch (dateTimeValues.getFilter()) {
+            case "Occoured On":
```

Comment:
Occur rather than Occour (if that is what it is elsewhere, that should probably be fixed there too)